File: server/src/app/bootstrap-node.ts
/**
 * Node.js 环境启动入口（用于测试）
 */

import { setRuntime } from '../framework/core/interfaces';
import { createNodeRuntime } from '../framework/runtime/node-adapter';

// 设置 Node.js 运行时
setRuntime(createNodeRuntime());

// 导入要测试的服务（它们会自动启动）
import './services/gateway';
import './services/login';
import './services/game';

console.log('========================================');
console.log('  Game Server Starting (Node.js Mode)  ');
console.log('========================================');

// Node.js 模式下，import 已经触发了 top-level 的 runtime.service.start()
// 因为 NodeService.start 是异步的 (setImmediate)，所以它们会排队启动

File: docker/lua/app/bootstrap-node.lua
local ____lualib = require("lualib_bundle")
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["5"] = 5,["6"] = 5,["7"] = 6,["8"] = 6,["9"] = 12,["10"] = 13,["11"] = 14,["12"] = 9,["13"] = 16,["14"] = 17,["15"] = 18});
local ____exports = {}
local ____interfaces = require("framework.core.interfaces")
local setRuntime = ____interfaces.setRuntime
local ____node_2Dadapter = require("framework.runtime.node-adapter")
local createNodeRuntime = ____node_2Dadapter.createNodeRuntime
require("app.services.gateway.index")
require("app.services.login.index")
require("app.services.game.index")
setRuntime(createNodeRuntime())
console:log("========================================")
console:log("  Game Server Starting (Node.js Mode)  ")
console:log("========================================")
return ____exports
